pkgs/dart_tooling_mcp_server/lib/src/mixins/dtd.dart

```diff
@@ -394,11 +408,43 @@ base mixin DartToolingDaemonSupport on ToolsSupport {
               ),
             ],
           );
-        } finally {
-          await vmService.callServiceExtension(
-            '$inspectorExtensionPrefix.disposeGroup',
+        }
+      },
+    );
+  }
+
+  /// Retrieves the selected widget from the currently running app.
+  ///
+  /// If more than one debug session is active, then it just uses the first one.
+  // TODO: support passing a debug session id when there is more than one debug
+  // session.
+  Future<CallToolResult> selectedWidget(CallToolRequest request) async {
+    return _callOnVmService(
+      callback: (vmService) async {
+        final vm = await vmService.getVM();
+        final isolateId = vm.isolates!.first.id;
+        try {
+          final result = await vmService.callServiceExtension(
+            '$_inspectorServiceExtensionPrefix.getSelectedSummaryWidget',
```

Comment:
How does this determine what things are "user code"? Do we need to call `addPubRootDirectories`? Or do we assume something else is?

Reply:
The Dart Code extension / Flutter IntelliJ plugin already do this today. Otherwise we assume the pub root directories is the root of the running app. CC @elliette to confirm this is how things still work today.

Reply:
Yes, that matches my understanding. Though it looks like the call to `addPubRootDirectories` was removed from the IntelliJ plugin: https://github.com/flutter/flutter-intellij/commit/1c1389fd10030487fa9802badcb64ca4aeb84688

FYI @pq @jwren this is another piece of code that might need to be added back